src/middlewared/middlewared/plugins/service.py

```diff
@@ -429,33 +428,23 @@ async def become_standby(self, service):
         service_object = await self.middleware.call('service.object', service)
         return await service_object.become_standby()
 
-    @accepts(Int("pid"), Int("timeout", default=10))
-    @returns(Bool(
-        "process_terminated_nicely",
-        description="`true` is process has been successfully terminated with `TERM` and `false` if we had to use `KILL`"
-    ))
-    def terminate_process(self, pid, timeout):
-        """
-        Terminate process by `pid`.
-
-        First send `TERM` signal, then, if was not terminated in `timeout` seconds, send `KILL` signal.
+    @private
+    def terminate_process(self, pid, timeout = 10):
         """
-        try:
-            process = psutil.Process(pid)
-            process.terminate()
-            gone, alive = psutil.wait_procs([process], timeout)
-        except psutil.NoSuchProcess:
-            raise CallError("Process does not exist")
+        Terminate the process with the given `pid`.
 
-        if not alive:
-            return True
+        Send SIGTERM, wait up to `timeout` seconds for the process to terminate.
+        If the process is still running after the timeout, send SIGKILL.
 
+        Returns:
+            boolean: True if process was terminated with SIGTERM, false if SIGKILL was used
+        """
+        if pid == 0 or pid == os.getpid():
```

Comment:
Looking at `kill(2)` man page I wonder whether we should also eliminate all negative values of `pid`.